### PR TITLE
Investigate prometheus #506

### PR DESCRIPTION
**Pull request by Buzzglo**
Initial code to start gathering metrics

### CODE REVIEW

**Comment by Buzzglo** [comment]
Could possibly require changes to this to implement events metrics. Pieter needs stats for the interim

File: src/gluon/commands/team/ListTeamMembers.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import {
     CommandHandler,
-    HandlerContext,
+    HandlerContext, logger,
```
Comment (KieranHons) [inline src/gluon/commands/team/ListTeamMembers.ts]:
```suggestion
    HandlerContext,
```

Comment (KieranHons) [reply src/gluon/commands/team/ListTeamMembers.ts]: No usage of logger within this class

File: src/gluon/metrics/prometheus/PrometheusClient.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    public static client: any;
+    public static guid: string;
+    public static counters: any[];
+
+    public static initialize(exp) {
+
+        PrometheusClient.client = require("prom-client");
+        // PrometheusClient.client.collectDefaultMetrics();
+
+        PrometheusClient.guid = Math.random().toString(36).substring(2)
```
Comment (KieranHons) [inline src/gluon/metrics/prometheus/PrometheusClient.ts]: Can we not use the uuid method here?

https://github.com/absa-subatomic/quantum-mechanic/blob/1e9d3c896c540eee347cf374acb6b02ac5cc7966/src/gluon/commands/packages/CreateApplicationProd.ts#L107

File: src/gluon/util/shared/BaseQMCommand.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import {
+    HandleCommand, HandlerContext, HandlerResult, logger, MappedParameter,
```
Comment (KieranHons) [inline src/gluon/util/shared/BaseQMCommand.ts]:
```suggestion
    HandleCommand,
    HandlerContext,
    HandlerResult,
    logger,
    MappedParameter,
```

File: src/gluon/commands/bitbucket/BitbucketProject.ts [hunk 4]
```diff
@@ -77,8 +71,11 @@ export class NewBitbucketProject extends RecursiveParameterRequestCommand
 
             await this.updateGluonWithBitbucketDetails(project.projectId, this.projectName, project.description, member.memberId);
 
-            return await success();
+            const result = await success();
+            this.succeedCommand();
+            return result;
```
[the remark this answers is not in the document]
Comment (Buzzglo) [reply src/gluon/commands/bitbucket/BitbucketProject.ts]: idk Andre only wanted to log a successCommand if it succeeded. I think a callback would be the way achieve this, but as you say this was a quick PR to get logs going. All of this should be cleaned up with the Events being added to the code.

File: src/gluon/metrics/prometheus/PrometheusClient.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export class PrometheusClient  {
+
+    public static client: any;
+    public static guid: string;
+    public static counters: any[];
+
+    public static initialize(exp) {
+
+        PrometheusClient.client = require("prom-client");
+        // PrometheusClient.client.collectDefaultMetrics();
```
[the remark this answers is not in the document]
Comment (andrejonathan) [reply src/gluon/metrics/prometheus/PrometheusClient.ts]: Done

File: src/gluon/commands/bitbucket/BitbucketProject.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (andrejonathan) [reply src/gluon/commands/bitbucket/BitbucketProject.ts]: All fixed for return await success()

File: src/gluon/commands/project/CreateProjectProdEnvironments.ts [hunk 6]
```diff
@@ -71,8 +65,11 @@ export class CreateProjectProdEnvironments extends RecursiveParameterRequestComm
 
             await this.gluonService.prod.project.createProjectProdRequest(member.memberId, project.projectId);
 
-            return success();
+            const result = success();
```
[the remark this answers is not in the document]
Comment (andrejonathan) [reply src/gluon/commands/project/CreateProjectProdEnvironments.ts]: All fixed for return await success()

File: src/gluon/commands/team/CreateTeam.ts [hunk 7]
```diff
@@ -48,8 +47,11 @@ export class CreateTeam implements HandleCommand<HandlerResult> {
 
             await this.createTeamInGluon(this.name, this.description, member.memberId);
 
-            return await success();
+            const result =  await success();
```
[the remark this answers is not in the document]
Comment (andrejonathan) [reply src/gluon/commands/team/CreateTeam.ts]: All fixed for return await success()

File: src/gluon/commands/bitbucket/BitbucketProjectAccessCommand.ts [hunk 8]
```diff
@@ -94,8 +89,11 @@ export class BitbucketProjectAccessCommand extends RecursiveParameterRequestComm
             }
             await taskRunner.execute(ctx);
 
-            return await success();
+            const result =  await success();
```
[the remark this answers is not in the document]
Comment (andrejonathan) [reply src/gluon/commands/bitbucket/BitbucketProjectAccessCommand.ts]: All fixed for return await success()

File: src/gluon/commands/project/NewProjectEnvironments.ts [hunk 9]
```diff
@@ -80,8 +74,11 @@ export class NewProjectEnvironments extends RecursiveParameterRequestCommand
 
             await this.requestProjectEnvironment(project.projectId, member.memberId);
 
-            return await success();
+            const result = await success();
```
[the remark this answers is not in the document]
Comment (andrejonathan) [reply src/gluon/commands/project/NewProjectEnvironments.ts]: All fixed for return await success()

File: src/gluon/commands/team/ListTeamMembers.ts [hunk 10]
```diff
@@ -48,30 +43,34 @@ export class ListTeamMembers extends RecursiveParameterRequestCommand
...
+                        color: "#c000ff",
+                        mrkdwn_in: ["text"],
+                    }],
+            };
 
-        this.succeedCommand();
-
-        return await ctx.messageClient.respond(msg);
+            this.succeedCommand();
+            return await ctx.messageClient.respond(msg);
```
[the remark this answers is not in the document]
Comment (andrejonathan) [reply src/gluon/commands/team/ListTeamMembers.ts]: Raised 
Fix explicit succeedCommands on return await ctx.messageClient #540
https://github.com/absa-subatomic/quantum-mechanic/issues/540

**Pull request merged**